### DISCUSSION TITLE
Why can't I get depth of field to work?

### DISCUSSION THREAD

**Discussion by jwwalker** [google/filament#3738] [Q&A] [answered]
I can see that depth of field works in the `gltf_viewer` sample, but I can't seem to get it to work in my own project.  I enable depth of field using `View::setDepthOfFieldOptions`, I set the camera aperture to f/1, set a large focal length, and set the focal distance to be very different (much smaller or much larger) than the actual distance from the camera to the object, but nothing gets fuzzy.   I also tried cranking up the `cocScale` parameter.  Any idea what I might be missing?  Or ideas on how to debug it?

**Comment by romainguy** [comment, marked as answer]
@pixelflinger

**Reply by jwwalker** [reply]
@romainguy Sorry, I figured it out.  For some dumb reason I had turned off post-processing, and apparently that is needed for DoF.

**Reply by romainguy** [reply]
Oh, yes indeed. Turning off post processing turns off many features. It's less useful to do so now that we offer a way to choose the tone mapper.
